Failure in test from ud_verbs/test_uct_perf

https://dev.azure.com/ucfconsort/0b36e3f0-8ab9-4a48-b68b-4b2350e02c88/_apis/build/builds/5894/logs/155
```
2020-05-05T13:18:07.1090549Z [----------] 1 test from ud_verbs/test_uct_perf
2020-05-05T13:18:07.1098392Z [ RUN      ] ud_verbs/test_uct_perf.envelope/1 <ud_verbs/mlx5_2:1>
2020-05-05T13:18:07.1664544Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:07.1926467Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:07.2123178Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:07.2398922Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:08.9838670Z [     INFO ] mlx5_2:1                am latency : 27066.898 usec (performance not checked)
2020-05-05T13:18:09.0179552Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:09.0422803Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:09.0585153Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:09.0808111Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:10.1447837Z [     INFO ] mlx5_2:1                   am rate : 0.000 Mpps (performance not checked)
2020-05-05T13:18:10.1703616Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:10.1940203Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:10.2134148Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:10.2365593Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:11.3010537Z [     INFO ] mlx5_2:1                 am rate64 : 0.000 Mpps (performance not checked)
2020-05-05T13:18:11.3255994Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:11.3494356Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:11.3664107Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:11.3879557Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:12.7998079Z [     INFO ] mlx5_2:1          am bcopy latency : 26699.299 usec (performance not checked)
2020-05-05T13:18:12.8249466Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:12.8551504Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:12.8668225Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:12.8905085Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:14.0962442Z [     INFO ] mlx5_2:1               am bcopy bw : 0.039 MB/sec (performance not checked)
2020-05-05T13:18:14.1218487Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:14.1465032Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:14.1672157Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:14.1864718Z ==25165== Warning: invalid file descriptor -1 in syscall close()
2020-05-05T13:18:15.0988345Z [swx-rdmz-ucx-new-02:25165:0:17339]  uct_tests.cc:596  Assertion `outstanding() == 0' failed
2020-05-05T13:19:09.0509554Z 
2020-05-05T13:19:09.0730962Z /scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/tools/perf/lib/uct_tests.cc: [ uct_perf_test_runner<(ucx_perf_cmd_t)0, (ucx_perf_test_type_t)1, (uct_perf_data_layout_t)2, false>::run_stream_req_uni() ]
2020-05-05T13:19:09.0949188Z       ...
2020-05-05T13:19:09.2465762Z       593         uct_perf_iface_flush_b(&m_perf);
2020-05-05T13:19:09.2668776Z       594         ucx_perf_get_time(&m_perf);
2020-05-05T13:19:09.2880694Z       595         ucs_assert(outstanding() == 0);
2020-05-05T13:19:09.3096473Z ==>   596         if (my_index == 1) {
2020-05-05T13:19:09.3314500Z       597             ucx_perf_update(&m_perf, 0, 0);
2020-05-05T13:19:09.3530624Z       598         }
2020-05-05T13:19:09.3739962Z       599 
2020-05-05T13:19:09.4366955Z 
2020-05-05T13:20:07.0912554Z ==== backtrace (tid:  17339) ====
2020-05-05T13:20:07.1122656Z  0 0x0000000000059478 ucs_debug_print_backtrace()  /scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/ucs/debug/debug.c:656
2020-05-05T13:20:07.1321276Z  1 0x00000000009d361e uct_perf_test_runner<(ucx_perf_cmd_t)0, (ucx_perf_test_type_t)1, (uct_perf_data_layout_t)2, false>::run_stream_req_uni()  /scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/tools/perf/lib/uct_tests.cc:596
2020-05-05T13:20:07.1517379Z  2 0x00000000009d361e uct_perf_test_runner<(ucx_perf_cmd_t)0, (ucx_perf_test_type_t)1, (uct_perf_data_layout_t)2, false>::run_stream_req_uni()  /scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/tools/perf/lib/uct_tests.cc:577
2020-05-05T13:20:07.1714425Z  3 0x00000000009bac0a uct_perf_test_runner<(ucx_perf_cmd_t)0, (ucx_perf_test_type_t)1, (uct_perf_data_layout_t)2, false>::run()  /scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/tools/perf/lib/uct_tests.cc:627
2020-05-05T13:20:07.1910386Z  4 0x00000000009b8eae ucx_perf_run()  /scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/tools/perf/lib/libperf.c:1807
2020-05-05T13:20:07.2105385Z  5 0x00000000005a0583 test_perf::thread_func()  /scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/test_perf.cc:170
2020-05-05T13:20:07.2302949Z  6 0x0000000000007dd5 start_thread()  pthread_create.c:0
2020-05-05T13:20:07.2697890Z  7 0x00000000000fdead __clone()  ???:0
2020-05-05T13:20:07.2895983Z =================================
2020-05-05T13:20:07.4554279Z [swx-rdmz-ucx-new-02:25165:0:17339] Process frozen...
2020-05-05T15:45:29.3873920Z ##[error]The operation was canceled.
2020-05-05T15:45:29.3890335Z ##[section]Finishing: Run ./contrib/test_jenkins.sh
```

--- a/src/uct/ib/ud/base/ud_iface.c
+++ b/src/uct/ib/ud/base/ud_iface.c
@@ -675,7 +675,8 @@ ucs_status_t uct_ud_iface_flush(uct_iface_h tl_iface, unsigned flags,
 
     uct_ud_enter(iface);
 
-    if (ucs_unlikely(uct_ud_iface_has_pending_async_ev(iface))) {
+    if (ucs_unlikely(uct_ud_iface_has_pending_async_ev(iface) ||
+                     !ucs_queue_is_empty(&iface->tx.outstanding_q))) {
         UCT_TL_IFACE_STAT_FLUSH_WAIT(&iface->super.super);
         uct_ud_leave(iface);
         return UCS_INPROGRESS;

